### DISCUSSION TITLE
Windows support

### DISCUSSION THREAD

**Discussion by PUFFINRUNNER** [gpiozero/gpiozero#1089] [Q&A]
Hey! Does you library work on Windows 10?

**Comment by PUFFINRUNNER** [comment]
What I meant to say is Windows 10 ARM version...

**Comment by PUFFINRUNNER** [comment]
Just read the article, I meant I'm willing to install the Windows 10 ARM version on Raspberry Pi 4 Model B, and I need to be able to use the GPIO pins

**Comment by bennuttall** [comment]
Have a read of https://gpiozero.readthedocs.io/en/stable/api_pins.html

You'd need to find a pin library that works on Windows

**Comment by waveform80** [comment]
Indeed -- `RPi.GPIO` relies upon the Linux sysfs file-system, and `lgpio` relies upon the Linux gpiochip devices so neither of these backends will work. gpiozero's `native` driver relies on the Linux `/dev/mem` or `/dev/gpiomem` devices (same as `RPi.GPIO`) so that won't work either. As Ben says, if you can find a Windows specific driver then it *should* be fairly simple to add a gpiozero pin driver for that, provided it supports all the required bits like pulls, edge detection and so forth.

Just to be clear: I've no objections to supporting Windows as well -- I did take care in things like `pinout` to ensure that it could still work on non-ANSI-supporting terminals like Windows' `cmd`. Unfortunately, I can't offer much in the way of help though; it's just not a system I've much familiarity with these days!
